src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -1294,6 +1295,26 @@ public ConsumerCloudData getConsumerCloudData() {
         return this.consumerCloudData;
     }
 
+    public boolean checkForCloudIdentifierFacts(Map<String, String> incomingFacts) {
```

Comment:
I think this method might make more sense as a private method in `ConsumerResource`. You could also add an additional parameter for the existing consumer facts so that you can use `Consumer.getFacts()` as the existingFacts parameter input. Also, I think we can name this method to indicate that it is checking for updates between the existing consumer facts and the provided consumer facts. Maybe something like `checkForCloudFactUpdates`?

`private boolean checkForCloudFactUpdates(Map<String, String> incomingFacts, Map<String, String> existingFacts)`

Reply:
This was one of the comments in design document from Nikos that we allready have similiar method so I should use similiar approach. https://github.com/candlepin/candlepin/blob/main/src/main/java/org/candlepin/model/Consumer.java#L560

Reply:
I proposed adding it to Consumer.java because there was a precedent already, but I suppose we could move both of these methods (`checkForCloudProfileFacts` and `checkForCloudIdentifierFacts`) into ConsumerResource, since it does make sense. 
Also, I think the word 'identifier' needs to stay in the new method name since it is clearer which facts we are talking about compared with the other method

Reply:
Ah that is my bad! I didn't see that there was already a method like this in Consumer.java. Knowing this now, I'm fine with keeping the two methods in Consumer.java or moving them to ConsumerResource, but we should do the same thing for both methods!

I also agree with Nikos on keep 'identifier' in the new method name.

Reply:
so how do we do it then? should I move it to ConsumerResource?

Reply:
Let's keep this method in Consumer.java.